_drafts/conviva-dag-process-time-prod-issue-draft.md

```diff
@@ -77,9 +77,17 @@ pub struct TypeRegistry<C: Clone, const N: usize = DEFAULT_N> {
 
 ## Post Mortem
 
-So, ArcSwap finally fixed the problem but it is worthwhile to understand why exactly that worked where concurrent hash maps had failed. It's worth repeating here again that our type registry, which is where the hash map was being used, was an *almost* read-only scenario.
+First, let's look a little more closely at how some concurrent hash maps typically operate. Many designs involve mechanisms like counters to track readers and writers, though the specifics can vary. For example, some implementations use a single, shared counter (not unlike a typical `RWLock`), while others employ sharded designs or multiple counters to reduce contention.
 
-First, let's look a little more closely at how concurrent hash maps typically operate (with a little hand-waving for simplicity's sake) - they use a *single, shared* counter which tracks how many readers are currently operating and allow a single writer to operate, not unlike a typical `RWLock`. 
+For instance, [Dashmap uses a sharded design](https://github.com/xacrimon/dashmap/blob/master/src/lib.rs#L85-L89) where each shard is a separate `HashMap` guarded by a `RWLock`
+
+```rust
+pub struct DashMap<K, V, S = RandomState> {
```

Comment:
Thanks, I think this is more than good enough.  I know you are concerned about going into detail about hashmaps and you don't have to even talk about how DashMap works.  I just wanted to address the accuracy of that statement about a single lock for most hashmaps.